slatedb/src/db_transaction.rs

```diff
@@ -909,4 +921,64 @@ mod tests {
             );
         }
     }
+
+    #[tokio::test]
+    async fn test_txn_scan_sees_concurrent_put_in_same_txn() {
```

Comment:
Nit: would be good to test that writes to an existing key are/aren't seen in these scenarios, too (vs. testing against a new key).

Reply:
Would a ref cell work here? 🤔 transactions aren't supposed to go between threads anyway...

Reply:
the tricky part involves with `async`, the `async` operations which held with `&self` reference to `DbTransaction` makes all the fields of `DbTransaction` to be alble to be enfored with Sync/Send 😭